src/providers/pkcs11/key_management.rs

```diff
@@ -228,38 +266,41 @@ impl Provider {
             return Err(ResponseStatus::PsaErrorInvalidArgument);
         }
 
-        template.push(Attribute::Class(ObjectClass::PUBLIC_KEY));
-        template.push(Attribute::KeyType(KeyType::RSA));
-        template.push(Attribute::Token(true.into()));
         template.push(Attribute::Modulus(modulus_object.into()));
         template.push(Attribute::PublicExponent(exponent_object.into()));
-        template.push(Attribute::Verify(true.into()));
         template.push(Attribute::Encrypt(true.into()));
-        template.push(Attribute::Id(key_id.to_be_bytes().to_vec()));
         template.push(Attribute::Private(false.into()));
         template.push(Attribute::AllowedMechanisms(vec![MechanismType::RSA_PKCS]));
+        template.push(Attribute::KeyType(KeyType::RSA));
 
-        trace!("CreateObject command");
-        match session.create_object(&template) {
-            Ok(key) => {
-                if let Err(e) =
-                    self.key_info_store
-                        .insert_key_info(key_triple, &key_id, key_attributes)
-                {
-                    format_error!("Failed to insert the mappings, deleting the key.", e);
-                    if let Err(e) = session.destroy_object(key) {
-                        format_error!("Failed to destroy public key: ", e);
-                    }
-                    Err(e)
-                } else {
-                    Ok(psa_import_key::Result {})
-                }
-            }
-            Err(error) => {
-                format_error!("Import key status: ", error);
-                Err(to_response_status(error))
-            }
+        Ok(())
+    }
+
+    pub(super) fn handle_ecc_public_import_attrib(
+        &self,
+        key_data: &[u8],
+        bits: usize,
+        curve_family: EccFamily,
+        template: &mut Vec<Attribute>,
+    ) -> Result<()> {
+        let mut data_in = key_data.to_vec();
+        if data_in.len() > 255 {
+            return Err(ResponseStatus::PsaErrorInvalidArgument);
         }
+        let mut ec_point = vec![0x04, data_in.len() as u8];
```

Comment:
The [description of the EC public key format](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_export_public_key.html#description) seems to imply that `0x04` is already included there? Also, is this valid for all curve families? Seems like it is all except Montgomery

edit: ah had another look, is it to have the same format as `[0x04, POINT_LENGTH, 0x04, X, Y]`? Which also explains the `> 255` check above.
Is it the format required by PKCS11? In the export function below you seem to say that there might be multiple.

I think a comment explaining all that would be nice here.

Reply:
Ok, yeah, I'll add a comment. From what I've read it seems those first two bytes are needed, though I tried initially without it and it worked. But I'm afraid some tokens will reject it. I should probs try this with some real hardware... 

And yes, I should probs check which curve is being imported.

Reply:
Changed, I found the proper representation, I just needed to wrap the PSA input data into an ASN.1 Octet String :)